javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -1095,7 +1095,7 @@ public int length() {
             init = init.put(init.size(), t);
             if (predicate.test(t)) {
                 if (init.size() == length()) {
-                    Tuple.of(this, empty());
+                    return Tuple.of(this, empty());
```

Comment:
Good finding! As far as I can see the semantic does not change because in the `==` case the loop ended and the fall-back return of the method does the same.